CMSgov/onemac
# Oy2 16332

Story: https://qmacbis.atlassian.net/browse/OY2-16332
Endpoint: https://d114ltsmatt982.cloudfront.net/

### Details

Creation of a left hand navigation element on a base/renewal waiver details page to show a table showing all temporary extensions

### Changes

- Added new left hand nav on details page
- created TempExtension component with an embedded Portal Table including actions
- added logic to getDetails to also include temp extension array for base/renewal waivers
- added seed data

### Implementation Notes

- Current implementation redeclares many pieces needed to implement the table actions. This is not ideal and should in the future be broken out to a separate component or some other implementation in order to foster reusability.

### Test Plan

1. Log in as a state submitter
2. Locate a Base or Renewal Waiver without any temp extensions
3. Navigate to the Waiver package detail view
4. Verify left hand nav has a "Temporary Extensions" item under "Package Overview" section
5. Click the Temporary Extension left hand nav item
6. Verify there is a table with columns "Extension Id","Status","Actions"
7. Verify when no temp extensions are on the waiver that the message "No currently submitted temporary extensions" displays under the table
8. Locate a Base or Renewal Waiver with a Temporary Extension Waiver in one of the following statuses: "In Review", "RAI Issued", or "Review Paused, Off the Clock" 
9. Navigate to the waiver details page and the Temporary Extension left hand nav item
10. Verify that the temp extension appears in the table
11. Verify the action column on the waiver has the Withdraw option
12. Verify that completing the withdraw action updates the temporary extension status to "Withdrawn"
